prow/apis/prowjobs/v1/types.go

```diff
@@ -501,6 +504,11 @@ const (
 // GCSConfiguration holds options for pushing logs and
 // artifacts to GCS from a job.
 type GCSConfiguration struct {
+	// Path is the path to upload to
+	// can be:
+	// * a local path: /path/to/storage
+	// * a remote path: e.g. gs://prow-artifacts or s3://prow-artifacts
+	Path string `json:"path,omitempty"`
```

Comment:
Why do we need this, I think this should be configured in the `Bucket` field or not?

Reply:
Depends on what our goal is that's this point, options:
* keeping the bucket field and changing the behaviour:
   * use the protocol the bucket starts with, e.g. s3://prow-artifacts => s3
   * defaulting to file if it starts with /
   * defaulting to gs:// if no prefix is discovered
* deprecating the bucket and similar fields and adding a path field

Reply:
I do not have objections against the idea of a new `Path` or `PathPrefix` field, but it is IMHO a change that is orthogonal to S3 support. It also has a huge blast radius as it basically requires adjustment of everything in Prow that somehow relies on knowing paths, so we should keep it out of the S3 support story and instead just start allowing a provider name prefix that gets defaulted to `gs://`.